clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -1076,7 +1076,7 @@ private void close(long timeout, TimeUnit timeUnit, boolean swallowException) {
 
         if (this.sender != null && this.ioThread != null && this.ioThread.isAlive()) {
             log.info("Proceeding to force close the producer since pending requests could not be completed " +
-                    "within timeout {} ms.", timeout);
+                    "within timeout {} ms.", timeUnit.toMillis(timeout));
```

Comment:
Good catch. The message above "Overriding close timeout {} ..." on line 1060 has the same bug. Maybe we can fix it as well? Also, maybe we should just store `timeUnit.toMillis(timeout)` in a variable since we use it several times?